src/soc/amd/common/block/spi/fch_spi_ctrl.c

```diff
@@ -217,76 +247,129 @@ static int fch_spi_flash_protect(const struct spi_flash *flash, const struct reg
 				 const enum ctrlr_prot_type type)
 {
 	int ret;
-	u32 reg32, rom_base, range_base;
-	size_t addr, len, gran_value, total_ranges, range;
+	u8 reg8;
+	u32 reg32, rom_base, flags;
+	size_t addr, len, gran_value, total_ranges, range, protected_ranges;
 	bool granularity_64k = true; /* assume 64k granularity */
+	struct flash_mmap_window mmap_window;
 
 	addr = region->offset;
 	len = region->size;
 
+	/* Multiple flash windows not supported right now */
+	if (spi_flash_get_mmap_windows(&mmap_window) != 1)
```

Comment:
Yup, but how can I know how many windows there are uphand? This function is just broken

Reply:
Add a comment and `die()` instead of returning? Of course the proper solution would be to fix that function, e.g. by taking additional argument with size of buffer.

Reply:
Very ugly, but should suffice for a couple of years: https://github.com/Dasharo/coreboot/pull/467/commits/4193b864973b40e070bde11e56c308266a81a6d6#diff-d64d04b9b3a31e1e003d182ea9a17f23987c627403f3f8482925643654525a01R254-R259

This function is used in conjunction with the coreboot tables, where you have just a pointer which increases with written data. Did not want, to mess with the function in X places now.